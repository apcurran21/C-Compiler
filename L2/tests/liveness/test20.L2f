(@false_loopy
	0
	%newVariableBx <- 1
	:false_start
	cjump %newVariableBx = 11 :false_done
	goto :false_notdone
	:false_notdone
	rdi <- %newVariableBx
	rdi *= 5
	%variableEU0HW83MOT <- %newVariableBx
	call print 1
	%newVariableBx <- %variableEU0HW83MOT
	%newVariableBx += 2
	goto :false_start
	:false_done
	return
)
