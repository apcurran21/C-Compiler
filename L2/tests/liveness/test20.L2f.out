(
(in
(r12 r13 r14 r15 rax rbp rbx)
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(%newVariableBx r12 r13 r14 r15 rbp rbx)
(%newVariableBx r12 r13 r14 r15 rbp rbx)
(%newVariableBx r12 r13 r14 r15 rbp rbx)
(%newVariableBx r12 r13 r14 r15 rbp rbx rdi)
(%newVariableBx r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rbp rbx rdi %variableEU0HW83MOT)
(r12 r13 r14 r15 rax rbp rbx %variableEU0HW83MOT)
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(%newVariableBx r12 r13 r14 r15 rbp rbx)
(%newVariableBx r12 r13 r14 r15 rbp rbx)
(%newVariableBx r12 r13 r14 r15 rbp rbx rdi)
(%newVariableBx r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rbp rbx rdi %variableEU0HW83MOT)
(r12 r13 r14 r15 rax rbp rbx %variableEU0HW83MOT)
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(%newVariableBx r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
